tensorflow/compiler/xla/service/gpu/gemm_rewriter.cc

```diff
@@ -1159,14 +1160,22 @@ class GemmRewriterVisitor : public DfsHloRewriteVisitor {
     config.set_epilogue(GemmBackendConfig::BIAS);
     std::vector<HloInstruction *> operands(gemm->operands().begin(),
                                            gemm->operands().end());
-    operands.push_back(bias);
-
+    if (gemm->custom_call_target() == kCublasLtMatmulF8CallTarget) {
+      // Matrix and vector bias don't co-exist for FP8 matmul.
+      if (config.beta() != 0.0) {
+      // Skip non-trivial case.
+        return true;
+      }
+      operands[2] = bias;
```

Comment:
I still don't see why we should we use operands[2] to be the vector bias, when we don't do so for non-FP8 matmuls. Why not have the same logic for Fp8 matmuls as we do for Fp16 matmuls, by having the matrix bias and vector bias be different parameters? Otherwise this complicates the logic later of having to detect whether operands[2] is a vector or matrix bias.